draft-ietf-quic-multipath.md

```diff
@@ -186,6 +186,17 @@ capitals, as shown here.
 We assume that the reader is familiar with the terminology used in
 {{QUIC-TRANSPORT}}. When this document uses the term "path", it refers
 to the notion of "network path" used in {{QUIC-TRANSPORT}}.
+In addition, we define the following terms:
+
+- Path Identifier (Path ID): An identifier that is used to identify 
+  a path in a QUIC connection at an endpoint. Path Identifier is used 
+  in multipath control frames (etc. PATH_ABANDON frame) to identify a path. 
+  Endpoint pre-allocates a Path Identifier field when it provide a new Connection ID
+  for its peer. When endpoints address a path in multipath control frames, 
+  it refers to the Path Identifier field of the destination Connection ID 
+  used for sending packets on that particular path. Note that the Path Identifier 
+  for initial path is always 0. 
```

Comment:
This should include a precise definition of what we mean by "path". Crucially, a "path" is **not** defined by the 5-tuple, since NAT rebindings change the 5-tuple without one of the peers knowing about it.

Instead, a new path is opened when a peer **intentionally** starts sending packets from a new address.

Reply:
I am implementing this right now. The good thing about the "unique path ID" is that the header now carries a clear intention. If the destination CID correspond to path X, then the intention was clearly to send the packet on path X -- even in cases when the CID is not exactly the same that was used for that path previously. This closes the ambiguity of "CID renewal and NAT Rebinding at the same time".

Reply:
Record in issue #291